ast-nodes/assignment.dart

```diff
@@ -43,6 +43,11 @@ class Assignment implements Statement {
   void checkSemantics() {
     lhs.checkSemantics();
     rhs.checkSemantics();
+    
+    if (lhs is Variable && lhs.scopeMark.resolve((lhs as Variable).name).scopeMark?.readOnly == true) {
```

Comment:
No need to compare to true, a Boolean is a Boolean)

Reply:
But there might be null, not Boolean, that's why I compare with true

Reply:
There can't be a `null` as long as we've correctly written the semantic check. A couple lines earlier, we have `lhs.checkSemantics()` to ensure that `lhs` really does resolve to some declaration.